.github/workflows/maven.yml

```diff
@@ -0,0 +1,93 @@
+# The MIT License
+#
+# Copyright 2005-2006 The Codehaus.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+name: Java CI
+
+on:
+  push:
+    branches: 
+      - '*'
+      - '!gh-pages'
+  pull_request:
+    branches:
+      - '*'
+      - '!gh-pages'
+
+jobs:
+  build:
+
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macOS-latest ]
+        java:
+          # Hotspot VM, the "default" VM
+          - version: 8
+            dist: adopt-hotspot
+          - version: 11
+            dist: adopt-hotspot
+          - version: 16
+            dist: adopt-hotspot
+          # Zulu VM, based on Hotspot
+          - version: 8
+            dist: zulu
+          - version: 11
+            dist: zulu
+          - version: 16
+            dist: zulu
+          # OpenJ9, an alternate JVM implementation.
+          - version: 8
+            dist: adopt-openj9
+          - version: 11
+            dist: adopt-openj9
+          - version: 16
+            dist: adopt-openj9
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up cache for ~./m2/repository
+        uses: actions/cache@v2.1.3
+        with:
+          path: |
+            ~/.m2/repository
+            !~/.m2/repository/org/codehaus/mojo/aspectj
+            !~/.m2/repository/dev/aspectj/
+          key: maven-${{ matrix.os }}-java${{ matrix.java }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-${{ matrix.os }}-java${{ matrix.java }}-
+            maven-${{ matrix.os }}-
+
+      - name: Set up JDK
+        uses: actions/setup-java@v2.1.0
+        with:
+          java-version: ${{ matrix.java.version }}
+          distribution: ${{ matrix.java.dist }}
+
+      - name: Build with Maven
+        run: mvn --show-version --errors --batch-mode clean install site
```

Comment:
I think, instead of `install` we might want to use `verify`.

Reply:
Also, `-Dinvoker.streamLogs=true` or `-Dinvoker.streamLogsOnFailures=true`. Your choice @kriegaex

Reply:
I used `verify` and `-Dinvoker.streamLogsOnFailures=true`, according to your suggestions.